ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx

```diff
@@ -0,0 +1,490 @@
+    const {
+      selectedDB,
+      selectedRP,
+      selectedMeasurement,
+      selectedFields,
+    } = this.state
+    const selectedBucket = formatDBwithRP(selectedDB, selectedRP)
+    const script = renderScript(
+      selectedBucket,
+      selectedMeasurement,
+      selectedFields
+    )
+
+    onAddToScript(script)
+    onSetIsWizardActive(false)
+  }
+
+  private fetchAndSetDBsToRPs = async () => {
+    const {source} = this.props
+
+    this.setState({
+      dbsToRPs: {},
+      dbsToRPsStatus: RemoteDataState.Loading,
+      selectedDB: null,
+      selectedRP: null,
+      measurements: [],
+      measurementsStatus: RemoteDataState.NotStarted,
+      selectedMeasurement: null,
+      fields: [],
+      fieldsStatus: RemoteDataState.NotStarted,
+      selectedFields: [],
+    })
+
+    let dbsToRPs
+
+    try {
+      dbsToRPs = await this.fetchDBsToRPs(source.links.proxy)
+    } catch {
+      this.setState({dbsToRPsStatus: RemoteDataState.Error})
+
+      return
+    }
+
+    const [selectedDB, selectedRP] = getDefaultDBandRP(dbsToRPs)
+
+    this.setState(
+      {
+        dbsToRPs,
+        dbsToRPsStatus: RemoteDataState.Done,
+        selectedDB,
+        selectedRP,
+      },
+      this.fetchAndSetMeasurements
+    )
+  }
+
+  private fetchAndSetMeasurements = async () => {
+    const {source} = this.props
+    const {selectedDB} = this.state
+
+    this.setState({
+      measurements: [],
+      measurementsStatus: RemoteDataState.Loading,
+      selectedMeasurement: null,
+      fields: [],
+      fieldsStatus: RemoteDataState.NotStarted,
+      selectedFields: [],
+    })
+
+    let measurements
+
+    try {
+      measurements = await this.fetchMeasurements(
+        source.links.proxy,
+        selectedDB
+      )
+    } catch {
+      this.setState({
+        measurements: [],
+        measurementsStatus: RemoteDataState.Error,
+        selectedMeasurement: null,
+      })
+
+      return
+    }
+
+    this.setState(
+      {
+        measurements,
+        measurementsStatus: RemoteDataState.Done,
+        selectedMeasurement: measurements[0],
+      },
+      this.fetchAndSetFields
+    )
+  }
+
+  private fetchAndSetFields = async () => {
+    const {source} = this.props
+    const {selectedDB, selectedMeasurement} = this.state
+
+    this.setState({
+      fields: [],
+      fieldsStatus: RemoteDataState.Loading,
+      selectedFields: [],
+    })
+
+    let fields
+
+    try {
+      fields = await this.fetchFields(
+        source.links.proxy,
+        selectedDB,
+        selectedMeasurement
+      )
+    } catch {
+      this.setState({
+        fields: [],
+        fieldsStatus: RemoteDataState.Error,
+        selectedFields: [],
+      })
+
+      return
+    }
+
+    this.setState({
+      fields,
+      fieldsStatus: RemoteDataState.Done,
+      selectedFields: [],
+    })
+  }
+}
+
+async function fetchDBsToRPs(proxyLink: string): Promise<DBsToRPs> {
```

Comment:
how do you feel about moving the functions outside the component to a utils file?